### ISSUE TITLE
discrepancy between MethylDackel and IGV PBAT-style data aligned with bwameth

### ISSUE THREAD

**Issue by avilella** [dpryan79/MethylDackel#50]
Hi,

I tried to used MethylDackel extract on a dataset aligned with bwameth which was prepped with a PBAT-style method, in the sense that the data aligns to the CTOT and CTOB strands.

We used to do this by aligning with `bismark --pbat`:
```
Option '--pbat': alignments to original strands (OT and OB) strands were ignored (i.e. not performed)
```
And then extract the methylation counts with bismark_methylation_extractor.

Currently, we are evaluating the use of bwa-meth followed by MethylDackel. But because of the way bwa-meth labels the strands and the PBAT nature of the data, MethylDackel seems to be counting on the basis of a directional OT/OB alignment, rather than a PBAT CTOT/CTOB alignment. I spotted this by the discrepancy between the IGV colouring and the MethylDackel counts, see below:

![screen shot 2017-07-20 at 16 01 26](https://user-images.githubusercontent.com/158007/28424422-8000c1ac-6d65-11e7-9307-49c6f447b561.png)

Long story short, the question is:

As a replacement of the bismark --pbat + bismark_methylation_extractor combo, how can I run bwa-meth + MethylDackel extract to obtain the counts as a PBAT CTOT/CTOB method?

**Comment by dpryan79** [comment]
At the moment I'm not sure that's possible (for what it's worth, PBAT was just coming out when this package was created). Can you point to any convenient (ideally smallish, but I'll take big and high quality over small and low quality) public PBAT datasets? I can then play around a bit and see what works reasonably (or add an option to MethylDackel if needed).

**Comment by avilella** [comment]
Here is an example file where the image above comes from:

https://dl.dnanex.us/F/D/bygGZ2v3Y8qq8Pp8YGFyKyp81bp4xbJ67Bzk618X/CEG60-53-C1-oxBS_S1_L001_R1_001.1.cutB.seqtk_revcom.u.bwameth.paired.primary.barcoded.2Df.sorted.bam
https://dl.dnanex.us/F/D/kB19gVB0qjpP236Jzv9KjxBgjk7g7XgB0PqqQkPv/CEG60-53-C1-oxBS_S1_L001_R1_001.1.cutB.seqtk_revcom.u.bwameth.paired.primary.barcoded.2Df.sorted.bam.bai

It's small in size but has high coverage concentrated in certain regions.

**Comment by dpryan79** [comment]
Wonderful, thanks!

**Comment by avilella** [comment]
The reference is the standard GRCh38Decoy from NCBI:
https://dl.dnanex.us/F/D/jzGpFXgf85p8XQB3JVYY9fxYkFXVgQqP062Jx8v4/Homo_sapiens_NCBI_GRCh38Decoy.genome.fa

**Comment by avilella** [comment]
Let me know if I can supply any more relevant details...

**Comment by dpryan79** [comment]
Will do. I'm creating the indices at the moment.

**Comment by dpryan79** [comment]
A small update, I have a modified version of bwameth with a `--pbat` option that seems to be producing good results (I'm testing with DRR019425). I've only tested paired-end reads so far and need to give single-end reads a go. Once those are working nicely too I'll make a pull-request on bwameth, at which point you should be able to get some reasonable results with it. MethylDackel itself doesn't need any modifications, you can give it the BAM file from bismark (after sorting) and if bwameth with `--pbat` is used the results should be pretty similar (I get correlation coefficients of ~90%, with bwameth producing a LOT more alignments in a fraction of the time).

**Comment by avilella** [comment]
Thanks, I can give it a try when there is a `bwameth --pbat` to try.

**Comment by dpryan79** [comment]
Please try the `pbat` branch with the `--pbat` option [from my fork here](https://github.com/dpryan79/bwa-meth/tree/pbat). That now seems to be working correctly for SE and PE data.

**Comment by dpryan79** [comment]
@avilella Did you have a chance to test that bwameth branch? If so, did it produce reasonable results?

**Comment by avilella** [comment]
I didn't have a chance yet.
